force-app/main/default/classes/RD2_ChangeHistoryService.cls

```diff
@@ -0,0 +1,170 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Recurring Donations
+* @description All services specific to Recurring Donation Change History
+*
+*/
+public inherited sharing class RD2_ChangeHistoryService {
+
+    /***
+    * @description Determines if Change History is enabled for the org
+    */
+    public Boolean isChangeHistoryEnabled {
+        get {
+            if (isChangeHistoryEnabled == null) {
+                isChangeHistoryEnabled = new RD2_Settings().isChangeHistoryEnabled();
+            }
+            return isChangeHistoryEnabled;
+        }
+        set;
+    }
+
+    /***
+    * @description Instance of ChangeHistoryCreator to create Change History without sharing
+    */
+    private ChangeHistoryCreator changeHistoryCreator {
+        get {
+            if (changeHistoryCreator == null) {
+                changeHistoryCreator = new ChangeHistoryCreator();
+            }
+            return changeHistoryCreator;
+        }
+        set;
+    }
+
+    private List<RecurringDonationChangeHistory__c> changeHistoryList = 
+        new List<RecurringDonationChangeHistory__c>();
+
+    /**
+     * @description Returns true if any of the key fields used to create Change History has been changed.
+     * @param updatedRd Changed Recurring Donation
+     * @param originalRd Old Recurring Donation
+     * @return Boolean
+     */
+    public void handleHistoryFieldChanged(RD2_RecurringDonation updatedRd, RD2_RecurringDonation originalRd) {
+        RecurringDonationChangeHistory__c changeHistory;
+
+        changeHistory = changeHistoryCreator.setChangedFields(changeHistory, updatedRd, originalRd);
+
+        if (changeHistory != null) {
+            changeHistoryList.add(changeHistory);
+        }
+    }
+
+    public List<RecurringDonationChangeHistory__c> getChangeHistoryRecords(){
```

Comment:
This method seems to only be called from within this class or from test methods. Can we change the private with @TestVisible annotation?